PSI PROBE Version 4.1.0  Error: java.lang.ClassNotFoundException: org.apache.openejb.loader.Files

[trace-20231213-141537-873.zip](https://github.com/psi-probe/psi-probe/files/13662040/trace-20231213-141537-873.zip)

Error: 	java.lang.ClassNotFoundException: org.apache.openejb.loader.Files
exception
org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is java.lang.NoClassDefFoundError: org/apache/openejb/loader/Files
at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1087)
at psiprobe.ProbeServlet.doDispatch(ProbeServlet.java:86)
at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965)
at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:515)
at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:583)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:212)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156)
at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:352)
at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:117)
at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:83)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:126)
at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:120)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:107)
at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:93)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
at org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter.doFilter(AbstractPreAuthenticatedProcessingFilter.java:150)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:117)
at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:225)
at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:190)
at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354)
at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156)
at com.opensymphony.sitemesh.webapp.SiteMeshFilter.obtainContent(SiteMeshFilter.java:189)
at com.opensymphony.sitemesh.webapp.SiteMeshFilter.doFilter(SiteMeshFilter.java:101)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:181)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:156)
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:598)
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
at psiprobe.Tomcat85AgentValve.invoke(Tomcat85AgentValve.java:38)
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:679)
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:765)
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:617)
at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:934)
at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2101)
at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
at java.lang.Thread.run(Thread.java:750)

Caused by: java.lang.NoClassDefFoundError: org/apache/openejb/loader/Files
at psiprobe.controllers.deploy.UploadWarController.handleRequestInternal(UploadWarController.java:189)
at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:177)
at psiprobe.controllers.deploy.UploadWarController.handleRequest(UploadWarController.java:57)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072)
... 56 more

Caused by: java.lang.ClassNotFoundException: org.apache.openejb.loader.Files
at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1338)
at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1150)
... 70 more

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,6 @@
+# Checkstyle
+.checkstyle
+
 # Eclipse
 .settings
 .classpath
@@ -7,15 +10,19 @@
 # Intellij
 .idea
 *.iml
+*.iws
+*.ipr
 
 # Maven
-target/
-
-# Project
-*.log
-psi-probe.iws
-psi-probe.ipr
 .mvn/wrapper/maven-wrapper.jar
 *.releaseBackup
 release.properties
+target/
+
+# PMD
+.pmd
+.pmdruleset.xml
+
+# Project
 .github/keys/
+*.log

--- a/psi-probe-core/pom.xml
+++ b/psi-probe-core/pom.xml
@@ -27,6 +27,7 @@
 
     <name>psi-probe-core</name>
     <description>PSI Probe Core - Core logic, data models, and controllers</description>
+    <url>https://github.com/psi-probe/psi-probe/</url>
 
     <scm>
         <connection>scm:git:ssh://git@github.com/psi-probe/psi-probe.git</connection>

--- a/psi-probe-core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java
+++ b/psi-probe-core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java
@@ -124,8 +124,10 @@ public class CopySingleFileController extends AbstractTomcatContainerController 
       logger.error("Could not process file upload", e);
       request.setAttribute("errorMessage", getMessageSourceAccessor()
           .getMessage("probe.src.deploy.file.uploadfailure", new Object[] {e.getMessage()}));
-      Files.delete(tmpFile.toPath());
-      tmpFile = null;
+      if (tmpFile != null) {
+        Files.delete(tmpFile.toPath());
+        tmpFile = null;
+      }
     }
 
     String errMsg = null;

--- a/psi-probe-core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java
+++ b/psi-probe-core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java
@@ -15,6 +15,7 @@ import com.google.common.base.Strings;
 import java.io.File;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
@@ -26,7 +27,6 @@ import javax.servlet.http.HttpServletResponse;
 import org.apache.catalina.Context;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.FilenameUtils;
-import org.apache.openejb.loader.Files;
 import org.apache.tomcat.util.http.fileupload.FileItem;
 import org.apache.tomcat.util.http.fileupload.FileItemFactory;
 import org.apache.tomcat.util.http.fileupload.disk.DiskFileItemFactory;
@@ -124,7 +124,7 @@ public class CopySingleFileController extends AbstractTomcatContainerController 
       logger.error("Could not process file upload", e);
       request.setAttribute("errorMessage", getMessageSourceAccessor()
           .getMessage("probe.src.deploy.file.uploadfailure", new Object[] {e.getMessage()}));
-      Files.delete(tmpFile);
+      Files.delete(tmpFile.toPath());
       tmpFile = null;
     }
 
@@ -196,7 +196,7 @@ public class CopySingleFileController extends AbstractTomcatContainerController 
       if (errMsg != null) {
         request.setAttribute("errorMessage", errMsg);
       }
-      Files.delete(tmpFile);
+      Files.delete(tmpFile.toPath());
     }
     return new ModelAndView(new InternalResourceView(getViewName()));
   }

--- a/psi-probe-core/src/main/java/psiprobe/controllers/deploy/UploadWarController.java
+++ b/psi-probe-core/src/main/java/psiprobe/controllers/deploy/UploadWarController.java
@@ -13,6 +13,7 @@ package psiprobe.controllers.deploy;
 import java.io.File;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.util.List;
 
 import javax.servlet.http.HttpServletRequest;
@@ -21,7 +22,6 @@ import javax.servlet.http.HttpServletResponse;
 import org.apache.catalina.Context;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.FilenameUtils;
-import org.apache.openejb.loader.Files;
 import org.apache.tomcat.util.http.fileupload.FileItem;
 import org.apache.tomcat.util.http.fileupload.FileItemFactory;
 import org.apache.tomcat.util.http.fileupload.disk.DiskFileItemFactory;
@@ -101,7 +101,7 @@ public class UploadWarController extends AbstractTomcatContainerController {
       logger.error("Could not process file upload", e);
       request.setAttribute("errorMessage", getMessageSourceAccessor()
           .getMessage("probe.src.deploy.war.uploadfailure", new Object[] {e.getMessage()}));
-      Files.delete(tmpWar);
+      Files.delete(tmpWar.toPath());
       tmpWar = null;
     }
 
@@ -186,7 +186,7 @@ public class UploadWarController extends AbstractTomcatContainerController {
       if (errMsg != null) {
         request.setAttribute("errorMessage", errMsg);
       }
-      Files.delete(tmpWar);
+      Files.delete(tmpWar.toPath());
     }
     return new ModelAndView(new InternalResourceView(getViewName()));
   }

--- a/psi-probe-core/src/main/java/psiprobe/controllers/deploy/UploadWarController.java
+++ b/psi-probe-core/src/main/java/psiprobe/controllers/deploy/UploadWarController.java
@@ -101,8 +101,10 @@ public class UploadWarController extends AbstractTomcatContainerController {
       logger.error("Could not process file upload", e);
       request.setAttribute("errorMessage", getMessageSourceAccessor()
           .getMessage("probe.src.deploy.war.uploadfailure", new Object[] {e.getMessage()}));
-      Files.delete(tmpWar.toPath());
-      tmpWar = null;
+      if (tmpWar != null) {
+        Files.delete(tmpWar.toPath());
+        tmpWar = null;
+      }
     }
 
     String errMsg = null;

--- a/psi-probe-rest/pom.xml
+++ b/psi-probe-rest/pom.xml
@@ -27,6 +27,7 @@
 
     <name>psi-probe-rest</name>
     <description>PSI Probe Rest - RESTful service endpoints.</description>
+    <url>https://github.com/psi-probe/psi-probe/</url>
 
     <scm>
         <connection>scm:git:ssh://git@github.com/psi-probe/psi-probe.git</connection>

--- a/psi-probe-tomcat10/pom.xml
+++ b/psi-probe-tomcat10/pom.xml
@@ -27,6 +27,7 @@
 
     <name>psi-probe-tomcat10</name>
     <description>PSI Probe Tomcat 10.1.x Adapter - Implementation of features specific to Apache Tomcat 10.1.x</description>
+    <url>https://github.com/psi-probe/psi-probe/</url>
 
     <scm>
         <connection>scm:git:ssh://git@github.com/psi-probe/psi-probe.git</connection>

--- a/psi-probe-tomcat11/pom.xml
+++ b/psi-probe-tomcat11/pom.xml
@@ -27,6 +27,7 @@
 
     <name>psi-probe-tomcat11</name>
     <description>PSI Probe Tomcat 11.0.x Adapter - Implementation of features specific to Apache Tomcat 11.0.x</description>
+    <url>https://github.com/psi-probe/psi-probe/</url>
 
     <scm>
         <connection>scm:git:ssh://git@github.com/psi-probe/psi-probe.git</connection>

